src/_pytest/config/findpaths.py

```diff
@@ -121,28 +121,34 @@ def determine_setup(inifile, args, rootdir_cmd_arg=None, config=None):
                 break
             except KeyError:
                 inicfg = None
-        rootdir = get_common_ancestor(dirs)
+        if rootdir_cmd_arg is None:
```

Comment:
Note: only the "is None" branch appears to be covered here: https://codecov.io/gh/pytest-dev/pytest/commit/247bdb2c0f5ba64161de7f24e958919dfe0e02ea#D1-124